pkg/sshagent/sshagent.go

```diff
@@ -201,7 +201,13 @@ func NewSource(paths []string) (*Source, error) {
 	if len(paths) == 0 {
 		socket = os.Getenv("SSH_AUTH_SOCK")
 		if socket == "" {
-			return nil, errors.New("$SSH_AUTH_SOCK not set")
+			return nil, errors.New("SSH_AUTH_SOCK not set in environment")
+		}
+		absSocket, err := filepath.Abs(socket)
+		if err != nil {
+			return nil, fmt.Errorf("evaluating SSH_AUTH_SOCK in environment: %w", err)
+		} else {
```

Comment:
Linter is not happy with this else block.

Reply:
Ah, right.  Reworking a couple of those.